python/feathub/table/table_descriptor.py

```diff
@@ -79,3 +79,21 @@ def get_feature(self, feature_name: str) -> Feature:
         :return: The feature with the given name.
         """
         pass
+
+    @abstractmethod
+    def get_bounded_view(self) -> TableDescriptor:
+        """
+        Gets a copy of self that is bounded. All sources of the return TableDescriptor
+        are bounded.
+
+        :return: A bounded table descriptor.
+        """
+        pass
+
+    @abstractmethod
+    def is_bounded(self, registry: Registry) -> bool:
```

Comment:
This method requires `registry` as input because `FeatureView` needs it to get around the problem described in FLINK-30078. 

Can we avoid complicating the public API due to a bug in Flink? For example, maybe we can just return `source.is_bounded()` here, and raise exception in `feature_view.build()` as appropriate.

We can also remove `_force_table_bounded` from e.g. `JoinTransform` by doing so.